metricsbp/baseplate_hooks.go

```diff
@@ -35,21 +38,23 @@ func (h CreateServerSpanHook) OnCreateServerSpan(span *tracing.Span) error {
 // ends, with success=True/False tags for the counter based on whether an error
 // was passed to `span.End` or not.
 type spanHook struct {
-	metrics *Statsd
+	metrics           *Statsd
+	prometheusMetrics *PrometheusMetrics
```

Comment:
From what I've seen, the "best practice" for injecting prometheus metrics (especially when there's just one, like "active_requests" is to pass in the `prometheus.Counter` -- that also could allow you to do the "active_requests" per-transport here potentially, using `prometheus.CounterVec.With` to pre-populate the label.

Leaving this out to just do in the per-transport section also sounds fine if you prefer.

Reply:
thanks for the details. sounds like i will close this PR and add active requests to the per transport PRs.